### ISSUE TITLE
Enhanced UPE - Payment gateway compatibility

### ISSUE THREAD

**Issue by FangedParakeet** [Automattic/woocommerce-payments#5027]
### Description

This is the continuation on the integration of Stripe's new UPE payment flows. Please see [this post](pc2DNy-2e2-p2) for more details. This WCPay compatibility issue is one of many to ensure that existing WCPay functionality is unaffected and remains fully operational after refactoring the UPE to follow the new payment flow. 

This issue is to ensure that there is no incompatibility with other payment gateways. This is essentially a duplicate of #4721, with respect to the new enhanced UPE implementation.   

This issue is currently blocked, as we await for Stripe to provide us access to their beta UPE release and while we wait to complete the PoC implementation of this new feature (#5019). 

These are the payment gateways we take care of:
- [eWAY](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-eway)
- [GoCardless](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-gocardless)
- [PayFast](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-payfast)
- [PayPal Advanced](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-paypal-advanced)
- [PayPal Checkout](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout)
- [PayPal Here](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-paypal-here)
- [PayPal Payments](https://href.li/?https://github.com/woocommerce/woocommerce-paypal-payments)
- [PayPal Powered by Braintree](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-paypal-powered-by-braintree)
- [PayPal Pro](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-paypal-pro)
- [PayPal Pro Hosted](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-paypal-pro-hosted)
- [Purchase Order Gateway](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-purchase-order)
[SnapScan](https://href.li/?https://github.com/woocommerce/woocommerce-gateway-snapscan)
- [Stripe](https://github.com/woocommerce/woocommerce-gateway-stripe) 

### Acceptance criteria

* Just double-check quickly that the checkout flow of these other payment gateways remain unaffected.

### Additional context

[Stripe UPE – Options to solve WooPay compatibility and WC Pay loss of control over the payments UI/UX](pc2DNy-2e2-p2)

**Comment by frosso** [comment]
Testing plan:
- Download plugin
- Enable them
- Add them to the checkout
- Perform sanity checks of the checkout exp.

**Comment by timur27** [comment]
I went through the above list of the plugins. I wasn't able to check two of them due to lack of credentials, but the rest of the successful tests made me confident enough to confirm that the deferred intent creation UPE isn't impacting any other plugins. 

The deferred intent UPE can be treated as a hybrid of multiple already existing gateway experiences. In terms of the user experience, it's identical to the split UPE, which was tested earlier as well in https://github.com/Automattic/woocommerce-payments/issues/4721. I clicked around and performed a few sanity checks and deferred intent UPE behaves the same way with other gateways from the above list, confirming the parity with the split UPE. In terms of the implementation, on the back-end it mostly uses the legacy card gateway, which was tested over the time as well. 

Having this summary as well as performed tests, we're good and thus I'm closing the issue.
